### PR TITLE
Remove version constraints of PyTorch and its related libraries.

### PR DESCRIPTION
**Pull request by toshihikoyanase**
PyTorch and its related libraries released binary packages for Python 3.8 (c.f., https://github.com/pytorch/pytorch/issues/29090#issuecomment-579527365). This PR removes the version constraints from setupy.py.

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@hvy Thank you for your review and your suggestion about Lightning.

Let me share the conclusion. The version constraint of `pytorch_lightning` was reverted by https://github.com/optuna/optuna/pull/893/commits/148102a9e5b44e2d8f537f120efb034a91fdae73 because it does not show the Python 3.8 support clearly. On the other hand, Ignite has supported Python 3.8 since [v0.3.0](https://github.com/pytorch/ignite/releases/tag/v0.3.0), and I kept the change of this PR as it is.

> Maybe we should wait with merging this PR until after the upcoming release since it's quite close (and examples are only run once a day).

Makes sense. Please merge after the release.

**Comment by hvy** [comment]
Sorry for the delayed comment but how about also including currently excluded tests/examples from the CircleCI configuration?

File: .circleci/config.yml [hunk 1]
```diff
@@ -302,7 +301,7 @@ jobs:
           <<: *examples
           environment:
             OMP_NUM_THREADS: 1
-            IGNORES: chainermn_.*|dask_ml_.*|keras_.*|pytorch_.*|tensorflow_.*|tfkeras_.*|fastai_.*
+            IGNORES: chainermn_.*|dask_ml_.*|keras_.*|tensorflow_.*|tfkeras_.*|fastai_.*
```
Comment (hvy) [inline .circleci/config.yml]: Does the PyTorch Lightning example pass?

**Comment by hvy** [comment]
Verified Python 3.8 unit tests and examples locally.

**Comment by toshihikoyanase** [comment]
@hvy Thank you for your careful review!

**Comment by hvy** [comment]
Sorry, could you fix the merge conflict? I'll go ahead and merge this PR afterwards.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**